lib/ddtrace/span.rb

```diff
@@ -39,15 +41,17 @@ def initialize(tracer, name, options = {})
       @parent_id = options.fetch(:parent_id, 0)
       @trace_id = options.fetch(:trace_id, @span_id)
 
+      @context = options.fetch(:context, nil)
+
       @meta = {}
       @metrics = {}
       @status = 0
 
       @parent = nil
       @sampled = true
 
-      @start_time = Time.now.utc
-      @end_time = nil
+      @start_time = nil # set by Tracer.start_span
```

Comment:
BREAKING CHANGE! Well, actually no, would break code relying on undocumented features and doing strange things, but this is still significant. I strongly believe is better to leave this unset, as this is a `start` time and not an `init` time. If it's not started, `start_time` is `nil`, period.